forage-android/src/main/java/com/joinforage/forage/android/core/ui/element/state/pan/PanEditTextState.kt

```diff
@@ -0,0 +1,41 @@
+package com.joinforage.forage.android.core.ui.element.state.pan
+
+import com.joinforage.forage.android.core.ui.element.state.EditTextState
+import com.joinforage.forage.android.core.ui.element.state.FocusState
+
+/**
+ * An interface that represents information that Forage gets from the card.
+ * This includes the [USState] that issued the card.
+ */
+interface DerivedCardInfo {
+    val usState: USState?
+}
+
+/**
+ * An interface that represents the state of a
+ * [ForagePANEditText][com.joinforage.forage.android.ui.ForagePANEditText] Element.
+ * @property derivedCardInfo The [DerivedCardInfo] for the card number submitted to the Element.
+ */
+interface PanEditTextState : EditTextState {
+    val derivedCardInfo: DerivedCardInfo // the interface not the DTO
```

Comment:
Are both the interface and the dto supposed to be public?

Reply:
Great question! Two notes here:

> Regarding the interface, `PanEditTextState`, being public
Since this interface get's exposed in [a few methods in `ForagePanElement`](https://app.graphite.dev/github/pr/teamforage/forage-android-sdk/267/FX-1460-Insert-ForageVaultElement-into-PIN-class-hierarchy#comment-PRRC_kwDOIQqcBs5h6gNZ), we unfortunately need to make this interface public.

> Regarding the `dto` being public
What I think may be relevant to point out is that we no longer have a `data class` that implements the `PanEditTextState`; instead `PanEditTextState.from(...)` (see below) returns an anonymous object. Returning an anonymous object that implements the `PanEditTextState` instead of defining a specific `data class` has the desirable affect of us not needing to add any DTO classes to the public surface area! But, lmk I am not speaking to your point as I could by misunderstanding.